### ISSUE TITLE
Searching using field names with a dot is not possible

### ISSUE THREAD

**Issue by wwindcloud** [RedisLabsModules/RediSearch#225]
Let say I have created an index and add a document

```
FT.CREATE index SCHEMA dc.title TEXT
FT.ADD index 123123 1 FIELDS dc.title "A Title"
```

I would then search for that title like this

```
FT.SEARCH index @dc.title:A Title
```

Doesn't work, it said syntax error on **title**. Am I doing something wrong or there's something to escape I am not aware of? Thanks.

**Comment by dvirsky** [comment]
ATM the only non alphanumeric character allowed in field names is `_`. However I don't see anything wrong with allowing other punctuation marks or allowing escaping unsupported ones. Do you see the need for things beyond dot, that are accepted in other databases? a comma doesn't make sense, maybe `:`?

**Comment by wwindcloud** [comment]
IMHO, the choice of characters allowed in the field names should not be arbitrarily restricted without a good reason. I know `:` is used for key prefix separation, space for argument separation, but that's pretty much it. Actually, if we have proper escape support, I don't see why we can't have colons or spaces in field names, or better yet full unicode support, and even field names in complete binary.

Some backstory why I am using dot. Well I was trying to index from a database with dcterms, prime metadata, and stuff and they are all using a dot to indicate schema.element.qualifier, I know it's easy to do convertion from `.` to `_`, but that will take some work. It would be great if redis support such characters and many others natively.

And yeah, full unicode or even binary support would be better, you can't expect people to always use alphanumeric field names, especially in non-English environment. Thanks.

**Comment by dvirsky** [comment]
I'll allow anything that goes into a token - unicode, ascii and **escaped** punctuation marks - as long as it doesn't create ambiguity.

**Comment by dvirsky** [comment]
Fixed in eca6cb48ddd8a52469bc601f07881a4f5aed6eb0
